src/test/OptionSettlement.t.sol

```diff
@@ -707,4 +864,41 @@ contract OptionSettlementTest is Test, NFTreceiver {
         engine.exercise(optionId, toExercise);
         vm.stopPrank();
     }
+
+    function _writeTokenBalance(address who, address token, uint256 amt) internal {
+        stdstore.target(token).sig(IERC20(token).balanceOf.selector).with_key(who).checked_write(amt);
+    }
+
+    // TODO consider better pattern to DRY up event definitions
```

Comment:
Had the same thought, but I've run into this with forge before. they recommend redefinition in test code: https://book.getfoundry.sh/cheatcodes/expect-emit. 

one possibility is to have the test contract implement (via revert()s) the interface which defines the events. its verbose but the bonus is that you'll get build time errors if the events change, rather than failing tests.

Reply:
Yeah, I do hear you. Redefinition isn't that bad. Your inheritance suggestion is interesting, it is better than altering production code to DRY up tests — e.g., I've tried putting events in a library or similar low-gas / no-gas approaches to DRY up definition between production and test code, but it doesn't seem a good trade-off bc then the production code itself is harder to read. Will leave as-is for now